mvrx/src/main/kotlin/com/airbnb/mvrx/BaseMvRxViewModel.kt

```diff
@@ -541,44 +542,49 @@ abstract class BaseMvRxViewModel<S : MvRxState>(
         deliveryMode: DeliveryMode,
         subscriber: (T) -> Unit
     ): Disposable {
-        if (lifecycleOwner == null) {
-            return observeOn(AndroidSchedulers.mainThread())
-                .subscribe(subscriber)
-                .disposeOnClear()
-        }
+        return observeOn(AndroidSchedulers.mainThread())
+            .resolveSubscription(subscriber, lifecycleOwner, deliveryMode)
+            .disposeOnClear()
+    }
 
-        @Suppress("UNCHECKED_CAST") val lifecycleAwareObserver = MvRxLifecycleAwareObserver(
-            lifecycleOwner,
-            deliveryMode = deliveryMode,
-            lastDeliveredValue = if (deliveryMode is UniqueOnly) {
-                if (activeSubscriptions.contains(deliveryMode.subscriptionId)) {
-                    throw IllegalStateException(
-                        "Subscribing with a duplicate subscription id: ${deliveryMode.subscriptionId}. " +
-                                "If you have multiple uniqueOnly subscriptions in a MvRx view that listen to the same properties " +
-                                "you must use a custom subscription id. If you are using a custom MvRxView, make sure you are using the proper" +
-                                "lifecycle owner. See BaseMvRxFragment for an example."
-                    )
-                }
-                activeSubscriptions.add(deliveryMode.subscriptionId)
-                lastDeliveredStates[deliveryMode.subscriptionId] as? T
-            } else {
-                null
-            },
-            onNext = Consumer { value ->
-                if (deliveryMode is UniqueOnly) {
-                    lastDeliveredStates[deliveryMode.subscriptionId] = value
-                }
-                subscriber(value)
-            },
-            onDispose = {
-                if (deliveryMode is UniqueOnly) {
-                    activeSubscriptions.remove(deliveryMode.subscriptionId)
+    private fun <T : Any> Observable<T>.resolveSubscription(
+        subscriber: (T) -> Unit,
```

Comment:
Small comment: I would keep the order of these params consistent with the above functions.

Reply:
Make sense.